src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy

```diff
@@ -0,0 +1,179 @@
+package org.prebid.server.functional.tests.pricefloors
+
+import org.prebid.server.functional.model.mock.services.floorsprovider.PriceFloorRules
+import org.prebid.server.functional.model.Currency
+import org.prebid.server.functional.model.response.auction.Bid
+import org.prebid.server.functional.model.response.auction.BidResponse
+import org.prebid.server.functional.model.response.auction.ErrorType
+import spock.lang.PendingFeature
+
+import java.math.RoundingMode
+
+import static org.prebid.server.functional.model.Currency.EUR
+import static org.prebid.server.functional.model.Currency.USD
+
+class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
+
+    private static final int FLOOR_VALUE_PRECISION = 4
+
+    @PendingFeature
+    def "PBS should update bidFloor, bidFloorCur for signalling when request.cur is specified"() {
+        given: "Default BidRequest with cur"
+        def bidRequest = bidRequestWithFloors.tap {
+            cur = [EUR]
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response"
+        def floorValue = randomFloorValue
+        def floorsResponse = PriceFloorRules.priceFloorRules.tap {
+            floorMin = floorValue
+            data.modelGroups[0].values = [(rule): floorValue]
+            data.modelGroups[0].currency = USD
+        }
+        floorsProvider.setResponse(bidRequest.site.publisher.id, floorsResponse)
+
+        when: "PBS processes auction request"
+        floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain bidFloor, bidFloorCur from floors provider"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        verifyAll(bidderRequest) {
+            imp[0]?.bidFloor == floorValue
+            imp[0]?.bidFloorCur == floorsResponse.data.modelGroups[0].currency
```

Comment:
No point in using null-safe operator here. If imp is null, so then imp[0] will return NPE exception. Check pls similar places.